launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/TTSServlet.java

```diff
@@ -0,0 +1,238 @@
+        } else if (Strings.isNullOrEmpty(podcastDescription)) {
+            throw new IllegalArgumentException("Unable to get Podcast Description. Please try again");
+        } else if (Strings.isNullOrEmpty(podcastText)) {
+            throw new IllegalArgumentException("Unable to get Podcast Text. Please try again");
+        } else if (Strings.isNullOrEmpty(podcastLanguage)) {
+            throw new IllegalArgumentException("Unable to get Podcast Language. Please try again");
+        }
+
+        // Search for key from given feed id
+        Key desiredFeedKey = KeyFactory.stringToKey(feedKey);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity desiredFeedEntity = null;
+        try {
+            desiredFeedEntity = datastore.get(desiredFeedKey);
+        } catch (EntityNotFoundException e) {
+            e.printStackTrace();
+            res.sendError(HttpServletResponse.SC_CONFLICT, "Unable to find given URL key, Please try again");
+            return;
+        }
+
+        // Turn the xml string back into an object
+        String xmlString = (String) desiredFeedEntity.getProperty(XML_STRING);
+        RSS rssFeed = null;
+        try {
+            rssFeed = TranslationServlet.XML_MAPPER.readValue(xmlString, RSS.class);
+        } catch (Exception e) {
+            res.sendError(HttpServletResponse.SC_CONFLICT, "Unable to translate. Try again");
+            return;
+        }
+
+        // Synthesize The podcast Text
+        String ttsFeedId = KeyFactory.keyToString(desiredFeedEntity.getKey()); // key for the entity that contains the
+                                                                               // channel to be modified
+        ByteString synthesizedMp3 = null;
+        try {
+            synthesizedMp3 = synthesizeText(podcastText);
+        } catch (Exception e) {
+            res.sendError(HttpServletResponse.SC_CONFLICT, "unable to create mp3 from request. Please try again.");
+            return;
+        }
+        Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+        String itemCount = String.valueOf(rssFeed.getChannel().getItems().size());
+        BlobId blobId = BlobId.of(BUCKET_NAME, ttsFeedId + itemCount);
+        BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
+
+        // create an uploadble array of bytes for blobstore
+        byte[] mp3Bytes = synthesizedMp3.toByteArray();
+        storage.create(blobInfo, mp3Bytes);
+
+        // Generate mp3 link
+        String mp3Link = TTS_BASE_URL + ttsFeedId + itemCount; // creates unique ID for each episode
+
+        rssFeed.getChannel().addItem(podcastTitle, podcastDescription, podcastLanguage, userEmail, mp3Link);
+
+        desiredFeedEntity.setProperty(XML_STRING, RSS.toXmlString(rssFeed)); // update existing entity's XML string
+
+        datastore.put(desiredFeedEntity);
```

Comment:
right now any user can update any feed if they make a post request with someone else's feed. we should compare emails before updating the feed